spec/services/invoices/preview_service_spec.rb

```diff
@@ -222,6 +222,357 @@
+                  expect(fixed_charge_fees.size).to be_zero
+
+                  # Only subscription fee
+                  expect(result.invoice.fees_amount_cents).to eq(6)
+                end
+              end
+            end
+
+            context "with pay_in_arrears fixed charge on subsequent invoice" do
+              let(:fixed_charge) do
+                create(
+                  :fixed_charge,
+                  plan:,
+                  add_on:,
+                  charge_model: "standard",
+                  pay_in_advance: false,
+                  prorated: false,
+                  units: 1,
+                  properties: {amount: "5"}
+                )
+              end
+
+              before do
+                fixed_charge
+
+                event_timestamp = subscription.started_at + 1.second
+                create(
+                  :fixed_charge_event,
+                  organization:,
+                  subscription:,
+                  fixed_charge:,
+                  units: fixed_charge.units,
+                  timestamp: event_timestamp,
+                  created_at: event_timestamp
+                )
+
+                # Create invoice_subscriptions to simulate that subscription has been invoiced before
+                # First invoice (subscription_starting)
+                create(
+                  :invoice_subscription,
+                  subscription:,
+                  invoicing_reason: :subscription_starting,
+                  from_datetime: subscription.started_at,
+                  to_datetime: subscription.started_at + 1.month,
+                  created_at: 1.month.ago
+                )
+
+                # Second invoice (regular billing)
+                create(
+                  :invoice_subscription,
+                  subscription:,
+                  invoicing_reason: :subscription_periodic,
+                  from_datetime: subscription.started_at + 1.month,
+                  to_datetime: subscription.started_at + 2.months,
+                  created_at: 15.days.ago
+                )
+              end
+
+              it "includes pay_in_arrears fixed charge on subsequent invoice" do
+                travel_to(timestamp + 1.month) do
+                  result = preview_service.call
+
+                  expect(result).to be_success
+                  expect(result.invoice.fees.size).to eq(2) # subscription + fixed_charge
+
+                  fixed_charge_fees = result.invoice.fees.select { |f| f.fee_type == "fixed_charge" }
+                  expect(fixed_charge_fees.size).to eq(1)
+
+                  fixed_charge_fee = fixed_charge_fees.first
+                  expect(fixed_charge_fee.fixed_charge).to eq(fixed_charge)
+                  expect(fixed_charge_fee.amount_cents).to eq(500) # $5 * 1 unit * 100
+                end
+              end
+            end
+
+            context "with multiple fixed charges" do
```

Comment:
if it's possible I'd ask also add a test for fixed charges with different payment schedule: pay in advance and pay in arrears